common/src/main/scala/uk/ac/wellcome/s3/S3ObjectStore.scala

```diff
@@ -57,20 +59,35 @@ class S3ObjectStore[T] @Inject()(
   s3Config: S3Config,
   keyPrefixGenerator: KeyPrefixGenerator[T]
 ) extends Logging {
-  def put(sourcedObject: T)(implicit encoder: Encoder[T]): Future[String] = {
+  def put(sourcedObject: T)(implicit encoder: Encoder[T]): Future[URI] = {
     val keyPrefix = keyPrefixGenerator.generate(sourcedObject)
+
     S3ObjectStore.put[T](s3Client, s3Config.bucketName)(keyPrefix)(
       sourcedObject)
   }
 
-  def get(key: String)(implicit decoder: Decoder[T]): Future[T] = {
-    S3ObjectStore.get[T](s3Client, s3Config.bucketName)(key)
+  def get(uri: URI)(implicit decoder: Decoder[T]): Future[T] = {
```

Comment:
There probably is an obvious answer but, if this only works with `S3Uri`, why don't we enforce it in the type signature? Like `def get(s3Uri: S3Uri) ...`